### DISCUSSION TITLE
File generation versus terminal output

### DISCUSSION THREAD

**Discussion by Robert-K-S** [openai/whisper#1321] [Q&A] [answered]
I notice that whisper tries to generate files and that the types of files to be generated can be adjusted with --output_format {txt,vtt,srt,tsv,json,all}, -f {txt,vtt,srt,tsv,json,all}.

What is the difference between these files and a file that is generated using the > character in the command line, e.g., using

`sudo -u www-data /home/username/.local/bin/whisper '/home/username/public_html/filename.mp3' --model tiny --language English > '/home/username/filename.txt'`

I ask because I wonder if under certain conditions whisper will not have the permissions to write its own output files (e.g., when running as user www-data, as in the above example) but would nevertheless be able to write terminal output to a file using the > character.

And if that is the case, I wonder whether one of the options for the --output_format command line switch should be "none".

**Comment by jongwook** [comment, marked as answer]
It doesn't support `--output-format none`, and likely we won't plan to -- if `www-user` can write to `/tmp`, you can specify something like `--output_dir /tmp/whisper`.

Alternatively, you can write a separate Python script doing something similar to the `cli()` function:

https://github.com/openai/whisper/blob/248b6cb124225dd263bb9bd32d060b6517e067f8/whisper/transcribe.py#L374-L453

but skip the part saving the outputs.
